pkg/authenticator/authenticator_factory.go

```diff
@@ -0,0 +1,30 @@
+package authenticator
+
+import (
+	"fmt"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/access_token"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/access_token/file"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/authenticator/config"
+	k8sAuthenitcator "github.com/cyberark/conjur-authn-k8s-client/pkg/authenticator/k8s"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+)
+
+func NewAuthenticator(conf config.Configuration) (Authenticator, error) {
```

Comment:
I don't want to hold up the merge on this, but Go style guides (e.g. https://go.dev/doc/effective_go#commentary) recommend that any exported names should have a preceding doc comment that begins with that name, e.g. 
```
// NewAuthenticator creates an instance of the Authenticator interface based on configured authenticator type.
```
(My IDE is running `golint` automatically, so when I write a file I see warnings pop up. BTW `golint` is deprecated, but there are replacements, e.g. `golangci-lint`.)